### ISSUE TITLE
Add a toggle to switch on/off Touch ID security

### ISSUE THREAD

**Issue by colourfreak** [AlphaWallet/alpha-wallet-android#2408]
A friend of mine has not completed the full backup flow. So he wrote down the seed phrase, but he did not encrypt it with Touch ID. So he asked:
- Where is a good place to do this now?

Let's add an item in the Settings menu (system section), where you will be able to enable Touch ID. 
<img width="750" alt="Screenshot 2022-02-11 at 16 13 26" src="https://user-images.githubusercontent.com/51817359/153617178-b921a4c1-0b47-4da8-b903-9cff1e2dbf73.png">
![Settings_Full](https://user-images.githubusercontent.com/51817359/153617198-75345f89-d674-4ab2-a36c-bb00674ad46b.jpg)

### Zeplin:
https://zpl.io/251gOEW

**Comment by colourfreak** [comment]
> Is this so you can have accounts that aren't locked, and the app won't nag you to lock them?
> 
> Or so you can unlock and lock wallets?

What user can do with a wallet (what interaction) if his wallet is not locked with Touch ID?

I can think of two situations.

1) I just have created a new wallet, backed up, but have not completed the whole flow (I wrote down the seed phrase on paper) and went back without confirming or locking my wallet with Touch ID
2) I was riding a bike with my smartphone stuck to the front. It was frustrating to unlock my phone with Touch ID over and over again. So I disabled it for the entire phone. Now, after the ride I have unlocked the AlphaWallet app. 

If I would have lost my phone right after 1/ or 2/ would my funds will be at risk?

**Comment by hboon** [comment]
I don’t think it works like @colourfreak described in the issue. Can we plan this out properly before we make any changes? Write out and describe what is the flow first?

**Comment by JamesSmartCell** [comment]
Yep good idea @hboon let's figure out how this should work and what restrictions.

@colourfreak as a first step would you like the ability to unlock a wallet? So eg, I have backed up the wallet (maybe) and locked it but decide I no longer want to keep using touch ID (etc), I then unlock that wallet. This would probably be done from the Wallet Management screen (hamburger button in wallet selection).

**Comment by colourfreak** [comment]
![image](https://user-images.githubusercontent.com/51817359/155549336-5df1edf3-7e43-45c9-80dc-186fa0f1285b.png)

**Comment by hboon** [comment]
I see now these are 2 separate things :)

A. The toggle in iOS Settings > Passcode / Touch ID is for controlling whether the app prompts for authentication when it's launched. It doesn't affect how the wallet is handled/accessed or backups.

B. As for backup status:

Not backed up | Key "not locked"
-|-
<img width="200" alt="Screenshot 2022-02-25 at 11 21 09 AM" src="https://user-images.githubusercontent.com/56189/155647656-5c64116f-a319-48b2-baa1-e748e6c066d5.png">|<img width="200" alt="Screenshot 2022-02-25 at 11 21 21 AM" src="https://user-images.githubusercontent.com/56189/155647672-1d8e653d-b72e-43aa-8bf6-c0d64f451e61.png">

I see that Android has this for backup status:

Not backed up | Key "not locked"
-|-
<img width='200' src='https://user-images.githubusercontent.com/56189/155648401-2c2ae74e-4642-48e8-b625-bda9a7cecb13.jpg'>|<img width='200' src='https://user-images.githubusercontent.com/56189/155648416-1bd66213-86e8-4332-b63d-c59b5b1597f2.jpg'>

So it looks like it's working correctly (as in I was concerned with the backup process), but the Android doesn't seem to show the little backup progress indicator any more (annotated):

<img width="200" alt="Screenshot 2022-02-25 at 11 21 09 AM copy" src="https://user-images.githubusercontent.com/56189/155648461-33effc51-f1df-47d7-b23b-1ab468e0538c.png">

It used to, I think?

(The "BACK UP WALLET 0x232" button background color is also off)

@colourfreak so, maybe we need to implement (A) and also to fix (B)? I can't find (B) in Zeplin though.

**Comment by JamesSmartCell** [comment]
It may be dependent on how Android uses the secure enclave - which is probably different from iOS. Once your keys are locked in the secure enclave (which is a separate processor to the main one) they are managed by the enclave and you negotiate with the enclave via the Android OS security system.

If you remove the lock from your phone, the OS will invalidate all the keys which were locked (that is Android OS security policy).

What I can do is add an individual unlock on the key management menu, that is move the passcode/face ID toggle you highlighted above to the menu you access from the hamburger menu on the wallets page. Unlocking all keys at once could be dangerous; but can be done.

@colourfreak what do you think?

**Comment by hboon** [comment]
@JamesSmartCell > What I can do is add an individual unlock on the key management menu, that is move the passcode/face ID toggle you highlighted above to the menu you access from the hamburger menu on the wallets page. Unlocking all keys at once could be dangerous; but can be done.

Just to clarify:

> A. The toggle in iOS Settings > Passcode / Touch ID is for controlling whether the app prompts for authentication when it's launched. It doesn't affect how the wallet is handled/accessed or backups.

i.e. it is just a simple check in iOS at launch to see if the user can use the app at all. It doesn't attempt to unlock any key/wallet

> B. As for backup status:

As for this ^, I think it's just a GUI regression. The text for Android reflects the state, it's just not presenting the correct GUI element

**Comment by JamesSmartCell** [comment]
Ah ok, so that toggle is just for app startup? Then it's easy to add.

**Comment by hboon** [comment]
Yeah, just for startup (and resumes).

It's just there in iOS because we inherited that functionality.
